services/web/server/src/simcore_service_webserver/workspaces/_groups_db.py

```diff
@@ -0,0 +1,162 @@
+from simcore_postgres_database.models.workspaces_access_rights import (
+    workspaces_access_rights,
+)
+from sqlalchemy import func, literal_column
+from sqlalchemy.sql import select
+
+from ..db.plugin import get_database_engine
+from .errors import WorkspaceGroupNotFoundError
+
+_logger = logging.getLogger(__name__)
+
+### Models
+
+
+class WorkspaceGroupGetDB(BaseModel):
+    gid: GroupID
+    read: bool
+    write: bool
+    delete: bool
+    created: datetime
+    modified: datetime
+
+
+## DB API
+
+
+async def create_workspace_group(
+    app: web.Application,
+    workspace_id: WorkspaceID,
+    group_id: GroupID,
+    *,
+    read: bool,
+    write: bool,
+    delete: bool,
+) -> WorkspaceGroupGetDB:
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(
+            workspaces_access_rights.insert()
+            .values(
+                workspace_id=workspace_id,
+                gid=group_id,
+                read=read,
+                write=write,
+                delete=delete,
+                created=func.now(),
+                modified=func.now(),
+            )
+            .returning(literal_column("*"))
+        )
+        row = await result.first()
+        return parse_obj_as(WorkspaceGroupGetDB, row)
+
+
+async def list_workspace_groups(
+    app: web.Application,
+    workspace_id: WorkspaceID,
+) -> list[WorkspaceGroupGetDB]:
+    stmt = (
+        select(
+            workspaces_access_rights.c.gid,
+            workspaces_access_rights.c.read,
+            workspaces_access_rights.c.write,
+            workspaces_access_rights.c.delete,
+            workspaces_access_rights.c.created,
+            workspaces_access_rights.c.modified,
+        )
+        .select_from(workspaces_access_rights)
+        .where(workspaces_access_rights.c.workspace_id == workspace_id)
+    )
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(stmt)
+        rows = await result.fetchall() or []
+        return parse_obj_as(list[WorkspaceGroupGetDB], rows)
+
+
+async def get_workspace_group(
+    app: web.Application,
+    workspace_id: WorkspaceID,
+    group_id: GroupID,
+) -> WorkspaceGroupGetDB:
+    stmt = (
+        select(
+            workspaces_access_rights.c.gid,
+            workspaces_access_rights.c.read,
+            workspaces_access_rights.c.write,
+            workspaces_access_rights.c.delete,
+            workspaces_access_rights.c.created,
+            workspaces_access_rights.c.modified,
+        )
+        .select_from(workspaces_access_rights)
+        .where(
+            (workspaces_access_rights.c.workspace_id == workspace_id)
+            & (workspaces_access_rights.c.gid == group_id)
+        )
+    )
+
+    async with get_database_engine(app).acquire() as conn:
+        result = await conn.execute(stmt)
+        row = await result.first()
+        if row is None:
+            raise WorkspaceGroupNotFoundError(
```

Comment:
can you please make the errors in this file bit more reusable?

Reply:
improved